### PR TITLE
make connectivity task have programmable timeouts based on type of telem...

### PR DESCRIPTION
**Pull request by brentpicasso**
...etry task

Wireless type telemetry tasks don't have a timeout... until we sunset the android specific app

### CODE REVIEW

File: src/logger/connectivityTask.c [hunk 1]
```diff
@@ -38,7 +38,7 @@
 #define IDLE_TIMEOUT							configTICK_RATE_HZ / 10
 #define INIT_DELAY	 							600
 #define BUFFER_SIZE 							1025
-#define DISCONNECT_TIMEOUT                      10000
+#define TELEMETRY_DISCONNECT_TIMEOUT            10000
```
Comment (stieg) [inline src/logger/connectivityTask.c]: Milliseconds I presume?

File: src/logger/connectivityTask.c [hunk 2]
```diff
@@ -107,6 +107,7 @@ static void createCombinedTelemetryTask(int16_t priority, xQueueHandle sampleQue
 		params->init_connection = &null_device_init_connection;
 		params->serial = SERIAL_TELEMETRY;
 		params->sampleQueue = sampleQueue;
+		params->connection_timeout = 0;
```
Comment (stieg) [inline src/logger/connectivityTask.c]: Why 0?  What is the significance of 0?

Comment (stieg) [inline src/logger/connectivityTask.c]: NVM... read value below.  Nice usage of 0 here.

**Comment by stieg** [comment]
Use of a uint prevents the negative timeout case.  Why do we want to be able to disable it (just curious)?

**Comment by stieg** [comment]
Otherwise looks pretty good to me.  I would say ship it pending answers to those questions.

File: src/logger/connectivityTask.c (hunk 1 above)
Comment (brentpicasso) [inline src/logger/connectivityTask.c]: correctamundo

**Comment by brentpicasso** [comment]
The legacy (current production) app does not send any kind of heartbeat, so the bluetooth connectivity task will time out every 10 seconds to 're establish connection'.  This is to prevent this from happening. In the future, we may re-enable timeout.

**Pull request merged**